### PR TITLE
Convert ch4 (container) from .rst to .md format

### PR DESCRIPTION
**Pull request by bocharsky-bw**

### CODE REVIEW

**Comment by weaverryan** [comment]
TutsHero is done for this chapter!

File: knpu/container.md [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+container. A lot of DI containers exist in PHP, but let's use Composer to grab the
+simplest one of all, called [Pimple][1]. Add a `require` key to `composer.json` to
+include the library:
+
+[[[ code('002395a9a7') ]]]
+
+Make sure you've [downloaded Composer][2], and then run `php composer.phar install`
+to download Pimple.
+
+***SEEALSE
```
Comment (weaverryan) [inline knpu/container.md:19]: SEEALSO :)

File: knpu/container.md [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+ask for the `mailer` service, it only creates it once. Each call returns 
+the original object:
+
+```php
+    $mailer1 = $container['mailer'];
+    $mailer2 = $container['mailer'];
+
+    // there is only 1 mailer, the 2 variables hold the same one
+    $willBeTrue = $mailer1 === $mailer2;
+```
```
Comment (weaverryan) [inline knpu/container.md:72]: Thanks for doing this :+1:  - but no indentation needed :)

Comment (weaverryan) [inline knpu/container.md:72]: ... and of course, we need to activate syntax highlight still - KnpLabs/KnpUniversity#414

File: knpu/container.md (hunk 1 above)
Comment (bocharsky-bw) [inline knpu/container.md:19]: Oops, thanks :)

File: knpu/container.md (hunk 2 above)
Comment (bocharsky-bw) [inline knpu/container.md:72]: I just keep it as is and didn't notice indentation... and I'll try to fix syntax highlight today.

Comment (weaverryan) [inline knpu/container.md:72]: No worries! I fixed it  - just wanted to point it, but mostly say thanks for adding the code block in the first place.

**Pull request merged**